cmd/tink-worker/worker/container_manager_test.go

```diff
@@ -0,0 +1,395 @@
+		delay:            delay,
+		statusCode:       statusCode,
+		err:              err,
+		waitErr:          waitErr,
+	}
+}
+
+func (c *fakeDockerClient) ContainerCreate(
+	_ context.Context, _ *containertypes.Config, _ *containertypes.HostConfig, _ *networktypes.NetworkingConfig, _ *specs.Platform, _ string) (containertypes.ContainerCreateCreatedBody, error) {
+	if c.err != nil {
+		return containertypes.ContainerCreateCreatedBody{}, c.err
+	}
+
+	return containertypes.ContainerCreateCreatedBody{
+		ID: c.containerID,
+	}, nil
+}
+
+func (c *fakeDockerClient) ContainerStart(_ context.Context, _ string, _ types.ContainerStartOptions) error {
+	if c.err != nil {
+		return c.err
+	}
+	return nil
+}
+
+func (c *fakeDockerClient) ContainerInspect(_ context.Context, _ string) (types.ContainerJSON, error) {
+	if c.err != nil {
+		return types.ContainerJSON{}, c.err
+	}
+	return types.ContainerJSON{}, nil
+}
+
+func (c *fakeDockerClient) ContainerWait(_ context.Context, _ string, _ containertypes.WaitCondition) (<-chan containertypes.ContainerWaitOKBody, <-chan error) {
+	respChan := make(chan containertypes.ContainerWaitOKBody)
+	errChan := make(chan error)
+	go func(e error) {
+		time.Sleep(c.delay)
+		if e != nil {
+			errChan <- e
+			return
+		}
+		respChan <- containertypes.ContainerWaitOKBody{
+			StatusCode: int64(c.statusCode),
+		}
+	}(c.waitErr)
+	return respChan, errChan
+}
+
+func (c *fakeDockerClient) ContainerRemove(_ context.Context, _ string, _ types.ContainerRemoveOptions) error {
+	if c.err != nil {
+		return c.err
+	}
+	return nil
+}
+
+func TestContainerManagerCreate(t *testing.T) {
+	cases := []struct {
+		name         string
+		workflowName string
+		action       *pb.WorkflowAction
+		containerID  string
+		registry     string
+		clientErr    error
+		wantErr      error
+	}{
+		{
+			name:         "Happy Path",
+			workflowName: "saveTheRebelBase",
+			action: &pb.WorkflowAction{
+				TaskName:    "UseTheForce",
+				Name:        "blow up the death star",
+				Image:       "yav.in/4/forestmoon",
+				Environment: []string{"MODE=insane", ""},
+				Volumes:     []string{"/tie-fighter/darth_vader:/behind_you"},
+				Pid:         "1",
+			},
+			containerID: "nomedalforchewie",
+			registry:    "rebelba.se",
+		},
+		{
+			name:         "create failure",
+			workflowName: "saveTheRebelBase",
+			action: &pb.WorkflowAction{
+				TaskName:    "UseTheForce",
+				Name:        "blow up the death star",
+				Image:       "yav.in/4/forestmoon",
+				Environment: []string{"MODE=insane", ""},
+				Volumes:     []string{"/tie-fighter/darth_vader:/behind_you"},
+				Pid:         "1",
+			},
+			containerID: "nomedalforchewie",
+			registry:    "rebelba.se",
+			clientErr:   errors.New("You missed the shot"),
+			wantErr:     errors.New("DOCKER CREATE: You missed the shot"),
+		},
+	}
+	logger, err := log.Init("github.com/tinkerbell/tink")
+	if err != nil {
+		t.Errorf("Error initializing logger: %w", err)
+		return
+	}
```

Comment:
you're probably better of using https://pkg.go.dev/github.com/packethost/pkg@v0.0.0-20211110202003-387414657e83/log#Test instead, it has the benefit of only printing output when `test.v` is enabled or if the test fails.

Reply:
nice